frame/staking-rewards/src/lib.rs

```diff
@@ -44,16 +42,53 @@ pub use pallet::*;
 
 #[frame_support::pallet]
 pub mod pallet {
-
-	use composable_traits::currency::{BalanceLike, CurrencyFactory};
-	use frame_support::{traits::UnixTime, PalletId};
+	use composable_support::{
+		abstractions::{
+			counter::Counter,
+			utils::{
+				decrement::SafeDecrement,
+				increment::{Increment, SafeIncrement},
+				start_at::ZeroInit,
+			},
+		},
+		math::safe::SafeArithmetic,
+	};
+	use composable_traits::{
+		currency::{BalanceLike, CurrencyFactory},
+		staking::RewardPoolConfiguration::RewardRateBasedIncentive,
+	};
+	use frame_support::{
+		pallet_prelude::*, traits::UnixTime, transactional, BoundedBTreeMap, PalletId,
+	};
+	use frame_system::pallet_prelude::*;
+	use sp_arithmetic::{traits::One, Permill};
+	use sp_runtime::traits::BlockNumberProvider;
+	use sp_std::collections::btree_map::BTreeMap;
+	use std::fmt::Debug;
 
 	use crate::{prelude::*, weights::WeightInfo};
+
 	#[pallet::event]
 	#[pallet::generate_deposit(pub fn deposit_event)]
-	pub enum Event<T: Config> {}
+	pub enum Event<T: Config> {
+		/// Pool with specified id `T::PoolId` was created successfully by `T::AccountId`.
+		RewardPoolCreated {
+			/// Id of newly created pool.
+			pool_id: T::PoolId,
+			/// Owner of the pool.
+			owner: T::AccountId,
+			/// End block
+			end_block: T::BlockNumber,
```

Comment:
end block of pool? Why we mix rate config with pool config? Pool = Rate + Rewards + Staking.  if to define rewards configuration State separate from Input, then each reward will have rate and end block. end block in Input is one, but is split per reward.

Reply:
My conceptual design does not separate between those things for simplicity, let's not over refactor things for now.
But I'm open to making restructures when we have the complete system built.

Reply:
in SE simplicity is SOLID. conceptual desing = code too.